src/Sylius/Bundle/CoreBundle/CatalogPromotion/CommandHandler/RemoveInactiveCatalogPromotionHandler.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\CatalogPromotion\CommandHandler;
+
+use Doctrine\ORM\EntityManager;
+use Sylius\Bundle\CoreBundle\CatalogPromotion\Command\RemoveInactiveCatalogPromotion;
+use Sylius\Bundle\CoreBundle\CatalogPromotion\Exception\InvalidCatalogPromotionStateException;
+use Sylius\Component\Core\Model\CatalogPromotionInterface;
+use Sylius\Component\Promotion\Model\CatalogPromotionStates;
+use Sylius\Component\Promotion\Repository\CatalogPromotionRepositoryInterface;
+
+final class RemoveInactiveCatalogPromotionHandler
+{
+    public function __construct(
+        private CatalogPromotionRepositoryInterface $catalogPromotionRepository,
+        private EntityManager $entityManager,
+    ) {
+    }
+
+    public function __invoke(RemoveInactiveCatalogPromotion $command): void
+    {
+        /** @var CatalogPromotionInterface|null $catalogPromotion */
+        $catalogPromotion = $this->catalogPromotionRepository->findOneBy(['code' => $command->code]);
+
+        if (null === $catalogPromotion) {
+            return;
+        }
+
+        if ($catalogPromotion->getState() !== CatalogPromotionStates::STATE_INACTIVE) {
```

Comment:
Following [Adam's comment ](https://github.com/Sylius/Sylius/pull/14270/files#r970407617), I believe **need** to keep the `Inactive` in the name of the class, as it's an unchangeable functionality of it. UNLESS (😄) we extract this condition into a separate service (`RemovedCatalogPromotionCriteriaChecker`?) and make it customisable 💥 Which I believe should be done to improve extendability, but also could be done as an improvement after the merge 🖖

Reply:
I left the handler name `RemoveInactiveCatalogPromotion` because I'm not sure how to implement it, more specifically, what the flow would be. Only one criteria checker per action or many. I thought of a tagged iterator ( for multi-check ) or additional custom stamps ( for single check). I need to discuss this more :)